lib/money.rb

```diff
@@ -41,9 +41,9 @@ class << self
     def default_currency=(val)
       block =
         if val.respond_to?(:call)
-          -> { Money::Currency.new(val.call) }
```

Comment:
I've changed to `define_method`. So why don't you comment in the last version? It'll be easy to understand.

I've dropped support for returning symbol from proc `Money.default_currency = -> { :usd }`, and encouraged to return currency object (`-> { Currency.new(:usd) }`). Now there is no implicit conversion: what you return is what you get.